jscomp/runtime/caml_format.ml

```diff
@@ -526,13 +526,13 @@ let caml_format_float fmt x =
         -> 
         (*  this will not work large numbers *)
         (* ("%3.10f", 3e+56, "300000000000000005792779041490073052596128503513888063488.0000000000") *)
-        s := Js_float.toFixed x prec 
+        s := Js_float.toFixedWithPrecision x ~digits:prec 
       | "g" -> 
         let prec = if prec <> 0 then prec else 1 in
-        s := Js_float.to_exponential x (prec - 1);
+        s := Js_float.toExponentialWithPrecision x ~digits:(prec - 1);
         let j = Bs_string.index_of !s "e" in 
-        let  exp = int_of_float @@ Js_float.of_any @@ Bs_string.slice_rest !s (j + 1)  in 
-        if exp < -4 || x >= 1e21 ||Bs_string.length (Js_float.toFixed x 0) > prec then 
+        let  exp = int_of_float @@ float_of_string @@ Bs_string.slice_rest !s (j + 1)  in 
```

Comment:
I could move the external in here and use that if you want. But if there's a subtle issue here, shouldn't there be a test for it that fails?